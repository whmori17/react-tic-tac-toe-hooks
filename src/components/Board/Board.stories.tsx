import * as React from 'react';
import { Board } from '.';
import { Moves } from '@customTypes/Move';

export default {
  title: 'Board',
};

const onClick = (i: number) => console.log(i);
const emptySquares: Moves = ['', '', '', '', '', '', '', '', ''];
const squares: Moves = ['X', '', '', 'O', '', 'X', '', '', ''];

export const empty = () => <Board {...{ onClick, squares: emptySquares }} />;
export const withMoves = () => <Board {...{ onClick, squares }} />;
