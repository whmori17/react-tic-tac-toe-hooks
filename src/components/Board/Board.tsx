import React from 'react';
import { Square } from '..';
import { Moves } from '@customTypes/Move';

export interface BoardProps {
  squares: Moves;
  onClick: (position: number) => void;
}

export const Board: React.FC<BoardProps> = ({ squares, onClick }) => {
  const boardRows = [
    [0, 1, 2],
    [3, 4, 5],
    [6, 7, 8],
  ];

  return (
    <div className="game-board">
      {boardRows.map((row, index) => (
        <div className="board-row" key={index}>
          {row.map(position => (
            <Square value={squares[position]} onClick={() => onClick(position)} key={position} />
          ))}
        </div>
      ))}
    </div>
  );
};

Board.displayName = 'Board';
