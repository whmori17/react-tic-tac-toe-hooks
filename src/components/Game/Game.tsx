import { FC } from 'react';
import * as React from 'react';
import { Board, GameMove } from '..';
import { GameInfo } from '@components/GameInfo';
import { useGame } from '@services/hooks';

export const Game: FC = () => {
  const { history, stepNumber, handleClick, status, jumpTo } = useGame();

  return (
    <div className="game">
      <Board squares={history[stepNumber]} onClick={(position: number) => handleClick(position)} />
      <GameInfo {...{ status }}>
        {history.map((_, step) => (
          <GameMove {...{ step, onClick: step => jumpTo(step) }} key={step} />
        ))}
      </GameInfo>
    </div>
  );
};

Game.displayName = 'Game';
