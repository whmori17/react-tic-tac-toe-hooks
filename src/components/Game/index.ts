export { Game, BoardSquares, GameState } from './Game';
