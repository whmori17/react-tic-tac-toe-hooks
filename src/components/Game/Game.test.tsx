import * as React from 'react';
import { shallow } from 'enzyme';
import { Game } from './Game';
import { BoardProps } from '@components/Board/Board';

describe('Game', () => {
  it('should render', () => {
    const wrapper = shallow(<Game />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should set X on clicked move', () => {
    const wrapper = shallow(<Game />);

    wrapper.find('Board').simulate('click', 3);

    const boardProps = (wrapper.find('Board').props() as unknown) as BoardProps;

    expect(boardProps.squares[3]).toBe('X');
    expect(wrapper).toMatchSnapshot();
  });

  it('should set X as firsat move adn O as second', () => {
    const wrapper = shallow(<Game />);

    wrapper.find('Board').simulate('click', 3);

    const boardProps = (wrapper.find('Board').props() as unknown) as BoardProps;

    expect(boardProps.squares[3]).toBe('X');

    wrapper.find('Board').simulate('click', 2);

    const boardProps2 = (wrapper.find('Board').props() as unknown) as BoardProps;

    expect(boardProps2.squares[2]).toBe('O');
    expect(wrapper).toMatchSnapshot();
  });
});
