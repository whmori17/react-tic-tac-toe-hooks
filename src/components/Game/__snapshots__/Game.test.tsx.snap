// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game should render 1`] = `
<div
  className="game"
>
  <Board
    onClick={[Function]}
    squares={
      Array [
        "",
        "",
        "",
        "",
        "",
        "",
        "",
        "",
        "",
      ]
    }
  />
  <Component
    moves={Array []}
    status=""
  />
</div>
`;

exports[`Game should set X as firsat move adn O as second 1`] = `
<div
  className="game"
>
  <Board
    onClick={[Function]}
    squares={
      Array [
        "",
        "",
        "O",
        "X",
        "",
        "",
        "",
        "",
        "",
      ]
    }
  />
  <Component
    moves={Array []}
    status=""
  />
</div>
`;

exports[`Game should set X on clicked move 1`] = `
<div
  className="game"
>
  <Board
    onClick={[Function]}
    squares={
      Array [
        "",
        "",
        "",
        "X",
        "",
        "",
        "",
        "",
        "",
      ]
    }
  />
  <Component
    moves={Array []}
    status=""
  />
</div>
`;
